File: sangsoo/0x06 queue/18258.py
import sys
from collections import deque
# sys.stdin = open("bojinput/10845.txt", "r")
input = sys.stdin.readline
n = int(input())
inlist = []
# for _ in range(n):
#     inlist.append(input().split())
q = deque()
for i in range(n):
    inlist = input().split()
    if inlist[0] == "push":
        q.append(inlist[1])
    elif inlist[0] == "pop":
        if q:
            print(q.popleft())
        else:
            print("-1")
    elif inlist[0] == "size":
        print(len(q))
    elif inlist[0] == "empty":
        if q:
            print("0")
        else:
            print("1")
    elif inlist[0] == "front":
        if q:
            print(q[0])
        else:
            print("-1")
    elif inlist[0] == "back":
        if q:
            print(q[-1])
        else:
            print("-1")

File: sangsoo/0x06 queue/18258.c
#include <stdio.h>
#include <string.h>
int queue[2000001];
int head = 0;
int tail = 0;

void push(int push_data){
    queue[tail++] = push_data;
}

int empty(){
    return head == tail;
}

int pop(){
    if(empty()){
        return -1;
    }
    return queue[head++];
}

int front(){
    if(empty()){
        return -1;
    }
    return queue[head];
}

int back(){
    if(empty()){
        return -1;
    }
    return queue[tail - 1];
}
int size(){
    return tail-head;
}


int main(){
    int N = 0, push_data = 0;
    char command[10];

    scanf("%d", &N);
    for (int i = 0; i < N; i++){
        scanf("%s", command);
        if(!strcmp(command, "push")){
            scanf("%d", &push_data);
            push(push_data);
        }
        else if(!strcmp(command, "pop")){
            printf("%d\n", pop());
        }
        else if(!strcmp(command, "empty")){
            printf("%d\n", empty());
        }
        else if(!strcmp(command,"size")){
            printf("%d\n", size());
        }
        else if(!strcmp(command,"front")){
            printf("%d\n", front());
        }
        else if(!strcmp(command, "back")){
            printf("%d\n", back());
        }
    }
    return 0;
}